L1Trigger/TrackerTFP/plugins/ProducerKFin.cc

```diff
@@ -0,0 +1,207 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
```

Comment:
I assume the stubs arriving from the SF are already ordered by track number. So ProducerKFin should just need to process these stubs, deduce the hit pattern by seeing which layers are on each track, apply truncation, and output.  I've not understood why it's bothering to look at the TTTrack collection, or why it moves the stubs first to a vector, then to a deque and then finally back to a vector again.

Reply:
ProducerKFin reads no stubs from SF it reads TTTracks from SFout.